File: game/scripts/vscripts/legacy/items/item_shield_orb/item_shield_orb.lua
item_shield_orb = class({})

function item_shield_orb:OnSpellStart()
    local damage_block = 16
    local caster = self:GetCaster()

    caster:AddNewModifier(caster, nil, "modifier_shield", { duration = 6.0, damage_block = damage_block })

    EmitSoundOn("Hero_Sven.GodsStrength.Attack", caster)
    EmitSoundOn("Hero_Abaddon.AphoticShield.Cast", caster)
end

File: src/vscripts/items/item_shield_orb.ts
import { CustomItem } from "../abilities/framework/custom_ability";
import { registerAbility } from "../lib/dota_ts_adapter";
import { ModifierShield } from "../modifiers/modifier_shield";

@registerAbility("item_shield_orb")
class ItemShieldOrb extends CustomItem {
    OnSpellStart() {
        const damage_block = 16;

        ModifierShield.apply(this.GetCaster(), this.GetCaster(), undefined, { duration: 6.0, damageBlock: damage_block });

        EmitSoundOn("Hero_Sven.GodsStrength.Attack", this.GetCaster());
        EmitSoundOn("Hero_Abaddon.AphoticShield.Cast", this.GetCaster());
    }
}
